### PR TITLE
Add Viewport (Room drag and drop)

### PR DESCRIPTION
**Pull request by mtwzim**
Viewport created to comply with the needs of issue #10 .

To use the room camera:
```
const room = Room.create(/* ... */);

app.stage.addChild(new RoomCamera(room));
```

Setting drag options:
```
const room = Room.create(/* ... */);

app.stage.addChild(new RoomCamera(room, /* ... */));
```

And using plugins:
```
const room = Room.create(/* ... */);

app.stage.addChild(new RoomCamera(room).pluginName1().pluginName2()); // You can use none, one or all the plugins
```

Plugins available:

- drag **(actived by default)**
- bounce
- clamp
- clampZoom
- decelerate
- follow
- mouseEdges
- pinch
- snap
- snapZoom
- wheel

### CODE REVIEW

File: src/objects/room/RoomCamera.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    // Calling drag
+
+    this.viewport.drag(dragOptions);
+
+    // Adding room to viewport
+
+    this.viewport.addChild(room);
+
+    // Adding viewport to container
+    this.addChild(this.viewport);
```
Comment (jankuss) [inline src/objects/room/RoomCamera.ts:77]: Avoid unnecessary comments. The code here speaks for it self.

**Review by jankuss — CHANGES_REQUESTED** [review]
Is it possible to disable scrolling behavior of that plugin?
When I scroll I don't want the camera to move.

Apart from that, is it necessary to expose all of the api of the Viewport plugin?
I think we should use the features we need internally in the `RoomCamera` class. And that's mostly just dragging and dropping and snapping back when the room is out of bounds.

File: example/src/index.ejs [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-        </div>
-    </body>
-</html>
\ No newline at end of file
+  <body>
+    <div id="container">
+      <canvas id="root"></canvas>
+    </div>
+  </body>
+</html>
```
[the remark this answers is not in the document]
Comment (mtwzim) [reply example/src/index.ejs]: How do you manage formatting? who formatted the file was the https://prettier.io/

[the remark this answers is not in the document]
Comment (jankuss) [reply example/src/index.ejs]: I added a `.prettierrc` into master, so everybody has the same formatter.

File: example/src/index.ts [hunk 3]
```diff
@@ -2,12 +2,11 @@ import * as PIXI from "pixi.js";
...
 import {
   Room,
-  FloorFurniture,
   Avatar,
   Shroom,
   loadRoomTexture,
-} from "@jankuss/shroom";
-import { DummyRoom } from "./DummyRoom";
+  RoomCamera,
+} from "../../src/index";
```
[the remark this answers is not in the document]
Comment (mtwzim) [reply example/src/index.ts:9]: Of course, sorry I was careless and I didn't know about the npm link, I didn't use it.

**Review by mtwzim — COMMENTED** [review]
Thanks for the code review :)

**Review by mtwzim — COMMENTED** [review]
Solved

**Review by mtwzim — COMMENTED** [review]
Resolving this code review

**Pull request merged**